QSB/TransformSync/PlayerHUDMarker.cs

```diff
@@ -0,0 +1,54 @@
+﻿using QSB.Events;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using UnityEngine;
+
+namespace QSB.TransformSync
+{
+    class PlayerHUDMarker : HUDDistanceMarker
+    {
+        private uint _netId = uint.MaxValue;
+        private bool _isReady;
+
+        protected override void InitCanvasMarker()
+        {
+            _markerRadius = 0.3f;
+
+            _markerTarget = new GameObject().transform;
+            _markerTarget.parent = transform;
+            // I'm not really sure why this has to be 20 instead of 2 (the player height in units).
+            // But that's the only way it looks right.
+            _markerTarget.localPosition = Vector3.up * 20;
+        }
+
+        public void SetId(uint netId)
+        {
+            _netId = netId;
+            _isReady = true;
+        }
+
+        protected override void RefreshOwnVisibility()
+        {
+            if (_canvasMarker != null)
+            {
+                _canvasMarker.SetVisibility(true);
+            }
+
+
+        }
+
+        void Update()
+        {
+            if (_isReady && PlayerJoin.PlayerNames.ContainsKey(_netId))
```

Comment:
What's the reason for this check here in Update? Can't the marker be made in SetId directly?

Reply:
If the reason is that it must wait a frame, there is this hack/workaround:

- Start a coroutine
- yield return new WaitOneFrame (or something like that)

Reply:
Ah I forgot about this. Let me check if I can make it start based on the dependency instead of waiting for a frame, otherwise I'll do what you suggest.